staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -320,19 +320,15 @@ func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) err
 	newDeltas := append(f.items[id], Delta{actionType, obj})
 	newDeltas = dedupDeltas(newDeltas)
 
-	_, exists := f.items[id]
-	if len(newDeltas) > 0 {
-		if !exists {
-			f.queue = append(f.queue, id)
-		}
-		f.items[id] = newDeltas
-		f.cond.Broadcast()
-	} else if exists {
-		// We need to remove this from our map (extra items
-		// in the queue are ignored if they are not in the
-		// map).
-		delete(f.items, id)
+	// newDeltas will always have at least one delta, if there is a case where
```

Comment:
The intent of this code was that the deduping function could evolve over time. So, although it works this way now, changing the code makes it hard to optimize the deduping function.

Reply:
Before suggesting this change I tried to think which implications it would have to remove this test. `dedupDeltas` may change over time but from a conceptual point of view, a dedup operation will only output an empty slice if the input is empty. This is the only asumption we are making for this change, and I think that the assumption is quite logical.

Reply:
Makes sense, did not consider that option. Will remove that part tomorrow.